cadquery/cq.py

```diff
@@ -2993,7 +2993,9 @@ def twistExtrude(
 
         r = Compound.makeCompound(shapes).fuse()
 
-        if combine:
+        if isinstance(combine, str) and combine == "cut":
```

Comment:
It seems like this section of code ends up being repeated quite a bit, but I'm not sure that there's an quick/efficient way to avoid the code duplication. Probably best just to leave it for now unless someone has an idea.

Reply:
I'm not sure to understand what we are talking about. The plan would be to move what to `_combineWithBase` ?

Reply:
There is a lot of repetitive code like:
```python
        if isinstance(combine, str) and combine == "cut":
            newS = self._cutFromBase(r)
        elif isinstance(combine, bool) and combine:
            newS = self._combineWithBase(r)
        else:
            newS = self.newObject([r])
```

it could be refactored to:

```python
return self._combineWithBase(r, combine)
```

(ofc you'd need to update `_combineWithBase` accordingly).

Reply:
Ok I think I've refactored `_combineWithBase` like you thought about it.
I'm proposing to add a breaking change to `text` that has both `cut` and `combine` kwarg. In my opinion we should get rid of the `cut` kwarg and keep the `combine` consistent with other methods. It would be fairly easy for user to make the change but would result in errors in model that still use it

Let me know if there is an agreement on this or not so I can make the change (or not) @adam-urbanczyk @jmwright @marcus7070

Reply:
Long term it is fine, but please use this for now: https://github.com/CadQuery/cadquery/blob/ea66ded172c2d6458a1209723546fb8778aa3060/cadquery/utils.py#L8.